hi-ml/src/health_ml/utils/reports.py

```diff
@@ -440,9 +484,26 @@ def render(self, save_html: bool = True) -> None:
 
         :param save_html: Whether to save the HTML to file
         """
-        subs: str = self.env.from_string(self.template).render(**self.render_kwargs)
+        new_env = self.env.from_string(self.template)
+        subs: str = new_env.render(**self.render_kwargs)
         self.report_html = subs
         # write the substitution to a file
         if save_html:
             with open(self.report_path_html, 'w') as f_path:
                 f_path.write(subs)
+
+    def zip_report_folder(self) -> Path:
+        """
+        Zip the report folder at the location '<report_folder>.zip', preserving the directory structure.
+        Returns the path to the zipped folder
+
+        :return: The path to the zipped folder
+        """
+        import zipfile
+        report_files = self.report_folder.rglob("*.*")
+        zipped_folder_path = self.report_folder.with_suffix(".zip")
+        with zipfile.ZipFile(zipped_folder_path, "w") as zipped_folder:
+            for report_file in report_files:
+                zipped_folder.write(report_file, arcname=report_file.relative_to(self.report_folder))
+        print(f"Zipped folder path: {str(zipped_folder_path)}")
```

Comment:
Should this be written to a logger instead?

Reply:
Potentially, but I propose to do this in a future PR!